### PR TITLE
 refactor: [Noon] add new field max_amount to mandate request 

### LINKED ISSUE #3445: [BUG] : Noon Api Contract Changes
### Bug Description

Noon API contract has changed. Therefore subscription payments is failing in Noon sandbox.

### Expected Behavior

Noon subscriptions work in sandbox

### Actual Behavior

Throwing Missing Field error

### Steps To Reproduce

Provide an unambiguous set of steps to reproduce this bug. Include code or configuration to reproduce, if relevant.

Make a Noon Card or Applepay subscription payment through hyperswitch

### Context For The Bug

_No response_

### Environment

Are you using hyperswitch hosted version? Yes/No
If yes, please provide the value of the `x-request-id` response header to help us debug your issue.

If not (or if building/running locally), please provide the following details:
1. Operating System or Linux distribution:
2. Rust version (output of `rustc --version`): ``
3. App version (output of `cargo r --features vergen -- --version`): ``
Sandbox

### Have you spent some time checking if this bug has been raised before?

- [X] I checked and didn't find a similar issue

### Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### Are you willing to submit a PR?

Yes, I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by AkshayaFoiger**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Resolves #3445

_Note
1.While connector create ensure `test_mode` is set to true
2.Ensure the currency passed is `AED` and country is `AE`_

## How did you test it?
1.Create a CIT with noon
 with 
```
  "connector_metadata": {
        "noon": {
            "order_category": "pay"
        }
    }
```
Screenshot 2024-01-12 at 2 16 26 PM

</n>
without order_category it should throw an error

Screenshot 2024-01-12 at 2 16 44 PM

2. Make a MIT

Screenshot 2024-01-12 at 2 02 35 PM

3.Test through stripe compatibility layer
CIT
Screenshot 2024-02-06 at 7 11 36 PM

MIT 
Screenshot 2024-02-06 at 7 11 30 PM

MIT with greater amount
Screenshot 2024-02-06 at 7 11 19 PM

4.Test Applepay mandates

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by ArjunKarthik — DISMISSED** [review]

File: crates/router/src/connector/noon/transformers.rs [hunk 1]
```diff
@@ -161,7 +161,7 @@ pub struct NoonPayPal {
 }
 
 #[derive(Debug, Serialize)]
-#[serde(tag = "type", content = "data")]
+#[serde(tag = "type", content = "data", rename_all = "UPPERCASE")]
```
Comment (sai-harsha-vardhan) [inline crates/router/src/connector/noon/transformers.rs:164]: Please make sure this won't affect any of the payment methods

File: crates/router/src/connector/noon/transformers.rs [hunk 2]
```diff
@@ -332,32 +333,33 @@ impl TryFrom<&types::PaymentsAuthorizeRouterData> for NoonPaymentsRequest {
...
+                    | Some(data_models::mandates::MandateDataType::MultiUse(Some(mandate))) => {
+                        conn_utils::to_currency_base_unit(mandate.amount, mandate.currency)
+                    }
+                    _ => Err(errors::ConnectorError::MissingRequiredField {
+                        field_name: "setup_future_usage.mandate_data.mandate_type",
+                    })
+                    .into_report(),
+                }?;
+
+                Ok::<NoonSubscriptionData, error_stack::Report<errors::ConnectorError>>(
```
Comment (sai-harsha-vardhan) [inline crates/router/src/connector/noon/transformers.rs:351]: Isn't this redundant here?

File: crates/router/src/connector/noon/transformers.rs (hunk 1 above)
Comment (AkshayaFoiger) [reply crates/router/src/connector/noon/transformers.rs:164]: It doesn't effect any other pmd

**Review by sai-harsha-vardhan — APPROVED** [review]

File: crates/router/src/connector/noon/transformers.rs (hunk 2 above)
Comment (AkshayaFoiger) [reply crates/router/src/connector/noon/transformers.rs:395]: This type annotation is required.

**Review by SamraatBansal — APPROVED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Pull request merged**